File: LeetCode/49_Group_Anagrams.js
var groupAnagrams = function(strs) {
    const g = {};
    for (let i = 0; i < strs.length; i++) {
      let ordered = strs[i].split('').sort().join('');
      // console.log(ordered)
      if (!(ordered in g)) {
        g[ordered] = [strs[i]]
        // console.log(g)
      }
      else {
        g[ordered].push(strs[i])
        // console.log(g)
      }
    };
    console.log(g)
    console.log(Object.values(g))
};

var strs = ["eat","tea","tan","ate","nat","bat"]
groupAnagrams(strs)

/**
Input: strs = ["eat","tea","tan","ate","nat","bat"]
Output: [["bat"],["nat","tan"],["ate","eat","tea"]]

 */

File: LeetCode/49_Group_Anagrams.py

def groupAnagrams(strs):
    g = {}
    for w in strs:
        x = "".join(sorted(w))
        # print(sorted(w))
        if x in g:
            g[x].append(w)
        else:
            g[x] = [w]
    print(g.values())
    print(list(g.values()))
    



strs = ["eat","tea","tan","ate","nat","bat"]
groupAnagrams(strs)